### PR TITLE
Fixing clz portability

### LINKED ISSUE #5: Clz function portability
I think it is okay to use builtin but they must have a generic clz function.

Moreover this function should be move in another namespace & file.

### PR DESCRIPTION
**Pull request by Lomadriel**
Fix #5

### CODE REVIEW

File: include/wfc/utility/math.hpp [hunk 1]
```diff
@@ -4,10 +4,17 @@
...
+#define WFC__has_builtin(x) __has_builtin(x)
+#endif
+
 namespace wfc
 {
 	namespace details
 	{
+#if WFC__has_builtin(__builtin_clz) && WFC__has_builtin(__builtin_clzl) && WFC__has_builtin(__builtin_clzll)
 		template <typename T>
 		T clz(T x) // FIXME
```
Comment (CBenoit) [inline include/wfc/utility/math.hpp]: What about that?

File: include/wfc/utility/math.hpp [hunk 2]
```diff
@@ -30,6 +37,27 @@ namespace wfc
 				static_assert(!std::is_same_v<T, T>);
 			}
 		}
+#else
+		template <typename T>
+		T clz(T x) // FIXME
```
Comment (CBenoit) [inline include/wfc/utility/math.hpp]: Is that still to fix?

File: include/wfc/utility/math.hpp [hunk 3]
```diff
@@ -4,10 +4,17 @@
 #include <limits>
 #include <type_traits>
 
+#ifndef __has_builtin
+	#define WFC__has_builtin(x) 0
+#else
+#define WFC__has_builtin(x) __has_builtin(x)
```
Comment (CBenoit) [inline include/wfc/utility/math.hpp:10]: Undefine macro at the end of file is good practice.

**Review by CBenoit — CHANGES_REQUESTED** [review]

**Comment by Lomadriel** [comment]
Fixed requested changes

File: include/wfc/utility/math.hpp (hunk 1 above)
Comment (Lomadriel) [reply include/wfc/utility/math.hpp]: I forgot to remove them

File: include/wfc/utility/math.hpp (hunk 3 above)
Comment (Lomadriel) [reply include/wfc/utility/math.hpp:10]: Added undef at the end of the file.

**Review by CBenoit — APPROVED** [review]

**Pull request merged**